src/test/java/org/opensearch/index/codec/customcodecs/CustomCodecTests.java

```diff
@@ -90,13 +93,31 @@ public void testZstd() throws Exception {
         assertEquals(DEFAULT_COMPRESSION_LEVEL, storedFieldsFormat.getCompressionLevel());
     }
 
+    public void testZstdWithCompositeIndex() throws Exception {
+        Codec codec = createCodecService(false, true).codec("zstd");
+        assertStoredFieldsCompressionEquals(Lucene101CustomCodec.Mode.ZSTD, codec);
+        Lucene101CustomStoredFieldsFormat storedFieldsFormat = (Lucene101CustomStoredFieldsFormat) codec.storedFieldsFormat();
+        assertEquals(DEFAULT_COMPRESSION_LEVEL, storedFieldsFormat.getCompressionLevel());
+        // assert docValues to be of compositeCodec's docValuesFormat
+        assert codec.docValuesFormat() instanceof Composite912DocValuesFormat;
```

Comment:
Filter codec does not expose delegate, so could not find a better way for assertion.